cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,60 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
     }
 }
 
+void CallProfiler::serialize(std::ostream & outstream) const
+{
+    outstream << R"({)";
+    outstream << R"("radix_tree": )";
+    serialize_radix_tree(outstream);
+    outstream << R"(})";
+}
+
+void CallProfiler::serialize_radix_tree(std::ostream & outstream) const
+{
+    outstream << R"({)";
+    outstream << R"("nodes": [)";
+    serialize_radix_tree_nodes(*(m_radix_tree.get_current_node()), true, outstream);
+    outstream << R"(],)";
+    outstream << R"("current_node": )" << m_radix_tree.get_current_node()->key() << R"(,)";
+    outstream << R"("unique_id": )" << m_radix_tree.get_unique_node();
+    outstream << R"(})";
+}
+
+// NOLINTNEXTLINE(misc-no-recursion)
+void CallProfiler::serialize_radix_tree_nodes(const RadixTreeNode<CallerProfile> & node, bool is_first_node, std::ostream & outstream) const
```

Comment:
It's more readable to use some comment showing examples of the serialized results.